### PR TITLE
Create Redux Toolkit slice for Stripe card payments

### PR DESCRIPTION
**Pull request by i-hardy**
## What are you doing in this PR?

This migrates our Stripe card payment state to Redux Toolkit for both the subscriptions and contributions checkouts.

I've connected the StripeForm component for subscriptions to the store for the first time, using [typed useDispatch and useSelector hooks](https://react-redux.js.org/tutorials/typescript-quick-start#define-typed-hooks), as this is the current recommended best practice for connecting components to the store.

I've also created some test store creation functions for use in component tests- as our real stores are singletons they persist updates between tests, which is obviously undesirable, so these helper functions use the existing store setup functions but create a fresh Redux instance each time.

[**Trello Card**](https://trello.com/c/FqPuVHER)

## Why are you doing this?

This is part of our ongoing Redux Toolkit migration work. The use of Redux hooks, as opposed to the `connect` function is also recommended by the Redux core team, so this is the approach we should try to use for store-connected components going forward.

### CODE REVIEW

File: support-frontend/assets/components/subscriptionCheckouts/stripeForm/stripeForm.tsx [hunk 1]
```diff
@@ -78,15 +83,23 @@ const marginTop = css`
...
-	 * State
+	 * Redux hooks
+	 */
+	const dispatch = useSubscriptionsDispatch();
+	const setupIntentClientSecret = useSubscriptionsSelector(
+		(state) => state.page.checkoutForm.payment.stripe.setupIntentClientSecret,
+	);
+	const recaptchaCompleted = useSubscriptionsSelector(
+		(state) => state.page.checkoutForm.recaptcha.completed,
+	);
```
Comment (i-hardy) [inline support-frontend/assets/components/subscriptionCheckouts/stripeForm/stripeForm.tsx:94]: Using multiple selectors is [recommended](https://redux.js.org/style-guide/#call-useselector-multiple-times-in-function-components) to minimise potential re-renders.

**Comment by i-hardy** [comment]
[🛠 Redux Toolkit migration - create Stripe card form reducer 🛠](https://trello.com/c/FqPuVHER/533-%F0%9F%9B%A0-redux-toolkit-migration-create-stripe-card-form-reducer-%F0%9F%9B%A0)

File: support-frontend/assets/pages/digital-subscription-checkout/components/digitalCheckoutForm.test.tsx [hunk 2]
```diff
@@ -55,6 +55,16 @@ describe('Digital checkout form', () => {
 
 	let initialState: unknown;
 	beforeEach(() => {
+		// For some inexplicable reason, executing the setup intent fetching thunk on render
+		// in the StripeForm component causes this test suite to hang indefinitely.
+		// cf. https://github.com/guardian/support-frontend/blob/ecd79a404c60fdb7d7ea19eb8a56075ad3084471/support-frontend/assets/components/subscriptionCheckouts/stripeForm/stripeForm.tsx#L211
+		// This does not happen in near-identical tests for the other product checkouts??
+		// The problem appears to be specifically to do with createAsyncThunk itself
+		window.guardian = {
```
Comment (michaelbjacobson) [inline support-frontend/assets/pages/digital-subscription-checkout/components/digitalCheckoutForm.test.tsx:63]: Weird! 🫤

**Review by michaelbjacobson — APPROVED** [review]
All looks great to me! 😎

**Pull request merged**